src/ogdc_runner/recipe/__init__.py

```diff
@@ -1,17 +1,20 @@
 from __future__ import annotations
 
-from pathlib import Path
-
+import fsspec
 import yaml
 
 from ogdc_runner.constants import RECIPE_CONFIG_FILENAME
 from ogdc_runner.models.recipe_config import RecipeConfig
 
 
-def get_recipe_config(recipe_directory: Path) -> RecipeConfig:
+def get_recipe_config(recipe_directory: str) -> RecipeConfig:
     """Extract config from a recipe configuration file (meta.yml)."""
-    with (recipe_directory / RECIPE_CONFIG_FILENAME).open() as config_file:
-        config_dict = yaml.safe_load(config_file)
+    recipe_path = f"{recipe_directory}/{RECIPE_CONFIG_FILENAME}"
+    try:
+        with fsspec.open(recipe_path, "rt") as config_file:
+            config_dict = yaml.safe_load(config_file)
+    except FileNotFoundError as err:
+        raise FileNotFoundError(f"File not found at: {recipe_path}") from err
```

Comment:
Is it worth catching and re-raising this exception like this? Is the existing exception that gets raised not informative enough? 

Typically if I were to catch an error like this to re-raise another, I would define a package-specific error to raise (e.g., maybe something like `OgdcRecipeNotFound`?)